src/mofaflex/_core/mofaflex.py

```diff
@@ -740,15 +790,42 @@ def _preprocess_options(self, *args: _Options):
 
     def _adjust_options(self, data: dict[dict[AnnData]]):
         # convert input arguments to dictionaries if necessary
+        guiding_vars_names = (
+            self._data_opts.guiding_vars_obs_keys.keys() if self._data_opts.guiding_vars_obs_keys else []
+        )
+        for guiding_var_name in guiding_vars_names:
+            if isinstance(self._data_opts.guiding_vars_obs_keys[guiding_var_name], str):
+                self._data_opts.guiding_vars_obs_keys[guiding_var_name] = dict.fromkeys(
+                    data.group_names, self._data_opts.guiding_vars_obs_keys[guiding_var_name]
+                )
+
         for opt_name, keys in zip(
-            ("weight_prior", "factor_prior", "nonnegative_weights", "nonnegative_factors"),
-            (data.view_names, data.group_names, data.view_names, data.group_names),
+            (
+                "weight_prior",
+                "factor_prior",
+                "nonnegative_weights",
+                "nonnegative_factors",
+                "guiding_vars_likelihoods",
+                "guiding_vars_scales",
+            ),
+            (
+                data.view_names,
+                data.group_names,
+                data.view_names,
+                data.group_names,
+                guiding_vars_names,
+                guiding_vars_names,
+            ),
             strict=True,
         ):
             val = getattr(self._model_opts, opt_name)
             if not isinstance(val, dict):
                 setattr(self._model_opts, opt_name, dict.fromkeys(keys, val))
 
+        for guiding_var_name in guiding_vars_names:
+            if self._model_opts.guiding_vars_scales[guiding_var_name] is None:
+                self._model_opts.guiding_vars_scales[guiding_var_name] = 1.0
```

Comment:
Easier and more obvious to the user to default to 1.0 in ModelOptions and remove this.